aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SignInChallengeCognitoActions.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.actions
+
+import aws.sdk.kotlin.services.cognitoidentityprovider.model.ChallengeNameType
+import com.amplifyframework.auth.cognito.AuthEnvironment
+import com.amplifyframework.auth.cognito.helpers.SRPHelper
+import com.amplifyframework.auth.cognito.helpers.SignInChallengeHelper
+import com.amplifyframework.statemachine.Action
+import com.amplifyframework.statemachine.codegen.actions.SignInChallengeActions
+import com.amplifyframework.statemachine.codegen.data.AuthChallenge
+import com.amplifyframework.statemachine.codegen.events.SignInChallengeEvent
+import com.amplifyframework.statemachine.codegen.events.SignInEvent
+
+object SignInChallengeCognitoActions : SignInChallengeActions {
+    override fun verifySignInChallenge(
+        event: SignInChallengeEvent.EventType.VerifyChallengeAnswer,
+        challenge: AuthChallenge
+    ) = Action<AuthEnvironment>("VerifySignInChallenge") { id, dispatcher ->
+        logger?.verbose("$id Starting execution")
+        val evt = try {
+            val username = challenge.username
+            var challengeResponses = mapOf("USERNAME" to username)
+
+            challenge.getChallengeResponseKey()?.also {
+                challengeResponses = challengeResponses.plus(it to event.answer)
+            }
+
+            // TODO: add attributes data from event to challenge response map
+
+            val secretHash = SRPHelper.getSecretHash(
+                username,
+                configuration.userPool?.appClient,
+                configuration.userPool?.appClientSecret
+            )
+            secretHash?.also { challengeResponses = challengeResponses.plus("SECRET_HASH" to secretHash) }
```

Comment:
enumerate this and other strings to be able to be used in tests?

Reply:
Done

Reply:
This should be let instead of also since you don't need the return value and should also be using it.

However, I think a simple null check may be more readable here. since you are not trying to act on something within secretHash, but instead just getting its value for a different variable.